artemis/db.py

```diff
@@ -376,6 +409,67 @@ def _get_task_deduplication_data(self, task: Task) -> str:
             }
         ).replace("\x00", " ")
 
+    def take_single_report_generation_task(self) -> Optional[ReportGenerationTask]:
+        with self.session() as session:
+            query = session.query(ReportGenerationTask).filter(
+                ReportGenerationTask.status == ReportGenerationTaskStatus.PENDING.value
+            )
+            if query.count() > 0:
+                return query[0]  # type: ignore
+        return None
+
+    def mark_report_generation_task_as_completed(
+        self,
+        task: ReportGenerationTask,
+        status: ReportGenerationTaskStatus,
+        output_location: Optional[str] = None,
+        error: Optional[str] = None,
+    ) -> None:
+        with self.session() as session:
+            task.status = status.value
+            if output_location:
+                task.output_location = output_location
+            if error:
+                task.error = error
+            session.add(task)
+            session.commit()
+
+    def create_report_generation_task(
+        self, tag: Optional[str], comment: Optional[str], language: Language, skip_previously_exported: bool
+    ) -> None:
+        with self.session() as session:
+            task = ReportGenerationTask(
+                tag=tag,
+                comment=comment,
+                language=language.value,
+                skip_previously_exported=skip_previously_exported,
+                status=ReportGenerationTaskStatus.PENDING,
+            )
+            session.add(task)
+            session.commit()
+
+    def get_report_generation_task(self, id: int) -> Optional[ReportGenerationTask]:
+        with self.session() as session:
+            try:
+                return session.query(ReportGenerationTask).get(id)  # type: ignore
```

Comment:
As above, you are returning None, so for code simplicity you can just pass `.first(<filter>)` which will reduce unnecesary code block